app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SalesController < ApplicationController
+
+  # Add a sale for a product
+  def new
+    @sale    = Sale.new
+    @vendor  = Vendor.find(params[:vendor_id]) 
+  end
+
+  def create
```

Comment:
Needs error checking such that you only create a sale when no product_id is provided.